### PR TITLE
Recurring payment journey

### PR DESCRIPTION
**Pull request by ScottDormand96**
https://eaflood.atlassian.net/browse/IWTF-3941

Uppdate acceptance tests for recurring payments

### CODE REVIEW

File: packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature [hunk 1]
```diff
@@ -0,0 +1,150 @@
+@browser
+Feature: I want to buy a recurring adult annual fishing
+
+  Background: Buy an adult fishing licence
+    Given  I am at the start of the purchase journey
+
+  Scenario: Scenario 1 - 12 Month Adult licence selecting 2 rod sea trout licence - Immediate start - Enter contact details
```
Comment (irisfaraway) [inline packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:7]: This scenario description is inaccurate – the journey below selects a salmon licence, but here it says "2 rod sea trout". This mismatch is the case for all of the new scenarios across the new files – some will say "salmon" and actually select a 2-rod coarse licence, etc. It's possible there are some other mismatches as well so think all of these will need to be checked and fixed.

Give me a nudge when this is done and I'll get back on the rest of the review.

File: packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    And I am buying a licence for myself
+    And I enter "Adult" "Salmon" as the name
+    And I am 7 days over my 17th birthday
```
Comment (irisfaraway) [inline packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:10]: Are 17 year olds allowed to set up a recurring payment?

**Review by irisfaraway — CHANGES_REQUESTED** [review]

File: packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature (hunk 2 above)
Comment (ScottDormand96) [reply packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:10]: I based this off function in gafl
export const validForRecurringPayment = permission => {
  const licenseeAge = moment().tz(SERVICE_LOCAL_TIME).diff(moment(permission.licensee.birthDate), 'years')
  return (
    process.env.SHOW_RECURRING_PAYMENTS?.toLowerCase() === 'true' &&
    permission.licenceLength === '12M' &&
    permission.isLicenceForYou &&
    licenseeAge > JUNIOR_MAX_AGE &&
    process.env.CHANNEL?.toLowerCase() !== 'telesales'
  )
}
JUNIOR_MAX_AGE = 16

File: packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature (hunk 1 above)
Comment (ScottDormand96) [reply packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:7]: looks ike were incorrect for others fie I haven't edited as we, wi go through them

File: packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature (hunk 2 above)
Comment (jaucourt) [reply packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:10]: I'm not sure this is a correct assumption; we charge for an adult licence in an angler's eighteenth year (i.e. it is on or after their seventeenth birthday). However, I'm not sure where we stand with offering recurring payments to someone who is, essentially, a minor, even if they won't be by the time the recurring payment comes into force. Let's pick this issue up with Melinda before we merge this PR.

Comment (jaucourt) [reply packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:10]: I've asked Melinda, although I'm unlikely to hear back before Wednesday I think

Comment (jaucourt) [reply packages/frontend-acceptance-tests/src/features/8 - Recurring Payments/1_adult_licence-12-months_immediate_start_recurring.feature:10]: Darren said we're in the clear on this, so we're ok to merge this PR now

**Review by jaucourt — APPROVED** [review]

**Review by irisfaraway — APPROVED** [review]

**Review by jaucourt — APPROVED** [review]

**Pull request merged**